jaqpotpy/datasets/dataset_base.py

```diff
@@ -1,36 +1,111 @@
+            raise TypeError("Either a DataFrame or a path to a file must be provided.")
+        elif (df is not None) and (path is not None):
+            raise TypeError("Either a DataFrame or a path to a file must be provided.")
+
+        if df is not None:
+            if not isinstance(df, pd.DataFrame):
+                raise TypeError("Provided 'df' must be a pandas DataFrame.")
+            else:
+                self._df = df
+                self.path = None
+        elif path is not None:
+            self.path = path
+            extension = os.path.splitext(self.path)[1]
+            if extension == '.csv':
+                self._df = pd.read_csv(path)
+            else:
+                raise ValueError("The provided file is not a valid CSV file.")
+            
+        if not(isinstance(y_cols, str) or 
+               (isinstance(y_cols, list) and all(isinstance(item, str) for item in y_cols))
+               ):
+               raise TypeError("y_cols must be provided and should be either" 
+                               "a string or a list of strings") 
+        
+        if not(isinstance(x_cols, str) or 
+              (isinstance(x_cols, list) and all(isinstance(item, str) for item in x_cols)) or
+              (isinstance(x_cols, list) and len(x_cols) == 0) or
+              (x_cols is None)):
+               raise TypeError("x_cols should be either a string, an empty list"
+                               "a list of strings, or None") 
+
+        #Find the length of each provided column name vector and put everything in lists
+        if isinstance(y_cols, str):
+            self.y_cols = [y_cols]
+            self.y_cols_len = 1
+        elif isinstance(y_cols, list) :
+            self.y_cols = y_cols
+            self.x_cols_len = len(y_cols)
+
+        if isinstance(x_cols, str):
+            self.x_cols = [x_cols]
+            self.y_cols_len = 1
+        elif isinstance(x_cols, list) :
+            self.x_cols = x_cols
+            self.x_cols_len = len(x_cols)
+        elif x_cols is None:
+            self.x_cols= []
+            self.x_cols_len = 0
+        
+        self.task = task
         self._dataset_name = None
-        self._df = None
-        self._x_cols_all = None
-        self.path = path
-        self.x_cols = x_cols
-        self.y_cols = y_cols
-        self._task = "regression"
-        self.featurizer = None
-        self._featurizer_name = None
-        self._external = None
+        self._y = None
+        self._x = None
+
+    @property
+    def df(self) -> pd.DataFrame:
+        return self._df
+
+    @df.setter
+    def df(self, value: pd.DataFrame):
+        if not isinstance(value, pd.DataFrame):
+            raise ValueError("The value must be a pandas DataFrame.")
+        self._df = value
+
+    @df.deleter
+    def df(self):
+        del self._df
 
     @property
     def task(self):
         return self._task
 
     @task.setter
     def task(self, value):
+        if value is None:
```

Comment:
Suggesting to reduce to a single if statement 

`if value is None or value.lower() not in ['regression', 'classification']:`